cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -169,16 +164,11 @@ fn extend_input_to_reg(ctx: Ctx, spec: InsnInput, ext_spec: ExtSpec) -> Reg {
     dst.to_reg()
 }
 
-fn lowerinput_to_reg_mem(ctx: Ctx, input: LowerInput) -> RegMem {
-    // TODO handle memory.
-    RegMem::reg(lowerinput_to_reg(ctx, input))
-}
-
 /// Put the given input into a register or a memory operand.
 /// Effectful: may mark the given input as used, when returning the register form.
 fn input_to_reg_mem(ctx: Ctx, spec: InsnInput) -> RegMem {
-    let input = ctx.get_input(spec.insn, spec.input);
-    lowerinput_to_reg_mem(ctx, input)
+    // TODO handle memory; merge a load directly, if possible.
```

Comment:
So a separate PR will do the same for x64?

Reply:
Indeed, that's my plan for today :-)